src/ArraySetops.chpl

```diff
@@ -68,18 +68,23 @@ module ArraySetops
     // sorts and removes all values that occur
     // more than once
     proc setxor1dHelper(a: [] ?t, b: [] t) {
-      var aux = radixSortLSD_keys(concatset(a,b));
+      const aux = radixSortLSD_keys(concatset(a,b));
+      const D = aux.domain;
 
-      var sliceComp = sliceTail(aux) != sliceHead(aux);
+      const sliceComp = aux[..D.high-1] != aux[D.low+1..];
       
       // Concatenate a `true` onto each end of the array
       var flag = makeDistArray((sliceComp.size + 2), bool);
+      const fD = flag.domain;
       
-      flag[0] = true;
-      flag[{1..#(sliceComp.size)}] = sliceComp;
-      flag[sliceComp.size + 1] = true;
+      flag[fD.low] = true;
+      flag[fD.low+1..#fD.high-1] = sliceComp;
```

Comment:
seems like the computation to make `sliceComp` should go here unless the const keeps the copy from being made...???

Reply:
Mike: I think you're saying we could write this as:

```chapel
flag[fD.low+1..#fD.high-1] = aux[..D.high-1] != aux[D.low+1..];
```

and eliminate `sliceComp` altogether?  If so, I agree that this seems preferable (should remove an explicit array temp, saving memory without any real negative impact on performance that I can anticipate).  We'd also need to replace the `sliceComp.size + 2` expression, but it looks like this should just be `aux.size + 1` if I'm keeping the logic correct?

Typing that up, I'm noticing that the `#fD.high-1` expression feels a little awkward because I think it presumes a relationship between fD's high bound and its size, which locks in a specific indexing scheme (?).  So I think better would be something like:

```chapel
flag[fD.low+1..fD.high-1] = aux[..D.high-1] != aux[D.low+1..];
```

if I've interpreted and translated that properly.

While looking at that and wondering about the relationship between `fD` and `D`, I think we could also turn those from `const` declarations (which will create a new distributed domain) to `const ref` declarations (which will refer to the existing distributed domain).  I suspect it would have a positive (although possible modest) performance benefit without any real downside.

Reply:
@mcdobe100 I'd like you to change the code using @bradcray suggestions.

Reply:
Alright, just put those in, thanks for the suggestions, makes sense to me

Reply:
:+1: 

If you haven't, I'd suggest re-running your perf tests to make sure none of our changes have undermined your original motivations for this PR (hopefully they'll only have improved them).

Reply:
Reran and updated the description with the new numbers. Saw slight improvements in both categories (speed and memory).

Reply:
@mcdobe100 @bradcray Good Work!, seems like the sorts are dominating the running time now. I'll merge at this point.